common/test/acceptance/pages/lms/dashboard.py

```diff
@@ -44,6 +68,60 @@ def _get_course_name(el):
 
         return self.q(css='section.info > hgroup > h3 > a').map(_get_course_name).results
 
+    def check_enrollment_mode(self, course_name, mode='verified'):
+        """Verify enrollment in a given mode.
+
+        Checks if the dashboard contains an enrollment in the provided mode for
+        the provided course name.
+
+        Arguments:
+            course_name (str): The name of the course whose mode should be checked.
+            
+        Keyword Arguments:
+            mode (str): The expected enrollment mode.
+
+        Returns:
+            A Boolean; whether an enrollment in the provided mode for the provided
+                course name was found on the dashboard.
+        """
+        # Filter elements by course name, only returning the relevant course item
+        course_listing = self.q(css=".course").filter(lambda el: course_name in el.text).results
+        
+        if course_listing:
+            # There should only be one course listing for the provided course name
+            banner_classes = course_listing[0].get_attribute('class')
+        else:
+            return False
+
+        return mode in banner_classes
```

Comment:
This method returns the classes on the `<article>` element corresponding to the targeted course listing. These classes are `course` and the name of the mode in which the user is currently enrolled ([reference](https://github.com/edx/edx-platform/blob/master/lms/templates/dashboard/_dashboard_course_listing.html#L49)). As such, this page object could return the name of a single mode (the one in which the user is currently enrolled), but not the names of all modes available for the course.

Would you be comfortable with this method becoming `get_enrollment_mode`, and having it return the name of the mode in which the user is currently enrolled for the given course?